--- a/src/backend_llvm.rs
+++ b/src/backend_llvm.rs
@@ -20,7 +20,9 @@ fn fibonacci(n: i64) -> i64 {
     let mut a = 0;
     let mut i = 2;
     let mut b = 1;
-    nígbà i <= n {
+    nígbà (i as u64) < (n as u64) && (a as u64) + (b as u64) <= (n as u64) {
         jẹ́ ara: i64 = a + b;
         a = b;
         b = ara;
@@ -28,7 +30,9 @@ fn fibonacci(n: i64) -> i64 {
     padà b;
 }

-iṣẹ́ main() -> i64 {
+iṣẹ́ main() -> i64 {
+    let n: i64 = 7; // Example input
     jẹ́risí fibonacci(0) == 0;
     jẹ́risí fibonacci(1) == 1;
     jẹ́risí fibonacci(n) == fibonacci(7);
